### ISSUE TITLE
Disable global shortcuts / shortcuts from parent components

### ISSUE THREAD

**Issue by LoganDark** [reaviz/reakeys#7]
## I'm submitting a...
<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Other... Please describe:
</code></pre>

## Current behavior
It's currently impossible to disable existing keybindings- for example, this:

```tsx
{
	ref: menuRef,
	name: '',
	keys: ['*'],
	callback: () => false
}
```

This won't prevent parents' shortcuts from triggering when their key combo is pressed on the keyboard. This may be a limitation of Mousetrap, or maybe I'm just not using it correctly.

## Expected behavior
I _should_ be able to disable all outside keybindings

## What is the motivation / use case for changing the behavior?
Headless UI uses a focus trap that will crash the application if you have more than one dialog open at a time. My context menu uses a focus trap, so if someone uses a keyboard shortcut to open a dialog, the application will crash and the entire tree will unmount. Not very fun.

**Comment by LoganDark** [comment]
Okay, so it looks like this is a bit of user error - `ref` only works if it is passed an object with a `current` field... which means I have to do this

```
	const [menu, setMenu] = useState<HTMLDivElement | null>(null)
	const menuRef = useMemo(() => ({current: menu}), [menu])
```

since I have a useEffect that depends on `menu`.

Now, I can use stopPropagation correctly and the keybinds all still _work_ - but `callback` is called twice with every keypress. What?

**Comment by LoganDark** [comment]
> Now, I can use stopPropagation correctly and the keybinds all still _work_ - but `callback` is called twice with every keypress. What?

This was a combination of two things:
- I forgot to use useCallback in a parent component, causing the hotkeys to get regenerated every render.
- All the hotkeys were hidden, and as a shortcut I used the same name, but for some reason that caused `useHotkeys` to stop functioning correctly, in the same manner that React needs unique `key`s on components in a list. The hotkeys wouldn't be unregistered properly, causing duplicate hotkeys to exist with even more of the same name, amplifying the issue.

In the end I was able to fix this. The full solution to suppressing outside keybinds:

1. Make sure `ref` is an object with a `current` property.
2. Add your own `keydown` handler to the element that calls `stopPropagation` on every event.
3. Use a focus trap and make the entire element focusable via `tabIndex={0}` to keep the `Mousetrap` and `stopPropagation` functional
